src/main/java/org/humancellatlas/ingest/project/Project.java

```diff
@@ -67,4 +64,18 @@ public SubmissionEnvelope getOpenSubmissionEnvelope(){
         }
         return null;
     }
-}
\ No newline at end of file
+
+    @JsonIgnore
+    public boolean removeSubmissionEnvelope(SubmissionEnvelope submissionEnvelope) {
+        if(!submissionEnvelope.isOpen())
+            throw new UnsupportedOperationException("Cannot delete submission if it is already submitted!");
+        boolean success = true;
+        for (File supplementaryFile: supplementaryFiles) {
+            if ( supplementaryFile.getSubmissionEnvelope().equals(submissionEnvelope)) {
+                success = supplementaryFiles.remove(supplementaryFile) && success;
```

Comment:
you could fail early if a supplementary file wasn't removed.

Reply:
I've changed this to check if submission is contained in project first.
No need to check the boolean result anymore.